platform/pulp/tasking/tasks.py

```diff
@@ -200,15 +201,26 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         with transaction.atomic():
             task_status = TaskStatus.objects.create(pk=inner_task_id, state=TaskStatus.WAITING,
                                                     group=group_id, **parent_arg)
-            for tag in tag_list:
+            for tag in tags:
                 task_status.tags.create(name=tag)
 
+        # Wrap celery signature into kwargs for _queue_reserved_task
+        async_args = tuple()
+        async_kwargs = {
+            "tags": tags,
+            "group_id": group_id,
+            "args": args,
+            "kwargs": kwargs,
+        }
+        async_kwargs.update(options)
+
         # Call the outer task which is a promise to call the real task when it can.
-        _queue_reserved_task.apply_async(args=[task_name, inner_task_id, resource_id, args, kwargs],
+        _queue_reserved_task.apply_async(args=[task_name, inner_task_id, resource_id, async_args,
```

Comment:
I think this should become:

`_queue_reserved_task.apply_async(args=(task_name, inner_task_id, resource_id, args,  kwargs), queue=RESOURCE_MANAGER_QUEUE)`

The change includes a switch from a list to a tuple. Also it ensures `args` and `kwargs` are passed as positional arguments to _queue_reserve_task which expects them as positional arguments.